unstructured/ingest/main.py

```diff
@@ -63,7 +65,7 @@ def cleanup(self):
     def _filter_docs_with_outputs(self, docs):
         num_docs_all = len(docs)
         docs = [doc for doc in docs if not doc.has_output()]
-        num_docs_to_process = len(docs)
+        num_docs_to_process = len(docs) if self.max_docs is None else min(len(docs), self.max_docs)
```

Comment:
the change should be something right before this line, something like:
```
if self.max_docs is not None:
    docs = docs[self.max_docs]
```
then no change to `num_docs_to_process` is needed